### PR TITLE
HoD Antagging

### PR DESCRIPTION
**Pull request by whateverusername0**
## About the PR

heads of departments lost their mindshield and brig access
in exchange they can roll antag now
CE, RD, CMO and QM.
HoP, security and captain are excluded for logical reasons.

## Why / Balance

it would be a fun experiment
also bringing heads back to the earth and removing their "alawys right" status

## Further changes
might need some adjustments in the space law to double jailtime for a crime that a head of department committed.

:cl:
- tweak: Removed mindshields from heads of departments.
- tweak: Removed brig access from heads of departments.
- add: Heads of departments can now become traitors/changelings/heretics.

## Summary by CodeRabbit

- **New Features**
	- Enhanced flexibility in role configurations by commenting out specific properties for various job roles, including `canBeAntag` and `Brig` access.
	- Introduced a new blacklist for the `HeadRev` role in the `Revolutionary` entity, excluding certain components.
	- Detailed role definitions for `NukeopsCommander`, `NukeopsMedic`, and `Nukeops` added to the `Nukeops` entity.
	- Improved the `Zombie` entity with briefing text and sound for the `InitialInfected` role.
	- Organized event scheduler entities into separate tables for better management of game rules.
	- Added a new `BasicRoundstartVariation` entity for variations at round start.
- **Bug Fixes**
	- Adjusted special abilities for roles by commenting out `AddImplantSpecial` entries, allowing for potential future modifications.

### CODE REVIEW

**Comment by Aineias1** [comment]
image

**Review by Piras314 — CHANGES_REQUESTED** [review]
as I said many times in Discord, the chance of them being antag must be much lower.

File: Resources/Prototypes/Roles/Jobs/Medical/chief_medical_officer.yml [hunk 1]
```diff
@@ -24,18 +24,18 @@
...
-  canBeAntag: false
+  #canBeAntag: false
   access:
   - Medical
   - Command
   - Maintenance
   - Chemistry
   - ChiefMedicalOfficer
-  - Brig
+  #- Brig
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Resources/Prototypes/Roles/Jobs/Medical/chief_medical_officer.yml:34]: heads of departments are not supposed to have brig access when they are able to become antagonists

File: Resources/Prototypes/Roles/Jobs/Engineering/chief_engineer.yml [hunk 2]
```diff
@@ -22,19 +22,19 @@
...
   - ChiefEngineer
   - Atmospherics
-  - Brig
+  #- Brig
   - Cryogenics
   special:
-  - !type:AddImplantSpecial
-    implants: [ MindShieldImplant ]
+  #- !type:AddImplantSpecial
+  #  implants: [ MindShieldImplant ]
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Resources/Prototypes/Roles/Jobs/Engineering/chief_engineer.yml:37]: heads of departments are supposed to have mindshields when they are not able to become antagonists

File: Resources/Prototypes/Roles/Jobs/Engineering/chief_engineer.yml [hunk 3]
```diff
@@ -22,19 +22,19 @@
   icon: "JobIconChiefEngineer"
   requireAdminNotify: true
   supervisors: job-supervisors-captain
-  canBeAntag: false
+  #canBeAntag: false
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Resources/Prototypes/Roles/Jobs/Engineering/chief_engineer.yml:25]: it already is. security should never become antagonists as their main purpose is to catch them.

**Comment by whateverusername0** [comment]
their chance to become antagonists is as low as it can get (e.g. 4 per 30 more antaggable people)
also i can't possibly make it without shoveling the entirety of antag selection system.
right now it's basically Random.Pick()

still for a weekly experiment this would be a learning experience
if we'll really need to lower the chance then i will work on it. rn? :goberserk:

**Comment by Piras314** [comment]
lower the chance for sure

**Comment by ElPidor** [comment]
roll CE 
Traitor
objectives 
Steal CE boots
Escape to CC 
Yay i rolled **job**

**Comment by whateverusername0** [comment]
> roll CE Traitor objectives Steal CE boots Escape to CC Yay i rolled **job**
the fact that this exists
image

**Comment by whateverusername0** [comment]
> lower the chance for sure

idk how am i going to lower their chance since their job role is counted as the same as a tider in antag selection
i tell you i'll need to redo that shit

**Comment by Piras314** [comment]
goob players are not competent to deal with hods other than hop and hos being antaggable just as much as a tider.

**Comment by BombasterDS** [comment]
> their chance to become antagonists is as low as it can get (e.g. 4 per 30 more antaggable people) also i can't possibly make it without shoveling the entirety of antag selection system. right now it's basically Random.Pick()
> 
> still for a weekly experiment this would be a learning experience if we'll really need to lower the chance then i will work on it. rn? :goberserk:

I don't think its that hard to make. weightedRandom prototype with jobname: weight and check for it on AntagSelect

**Comment by JoeHammad1844** [comment]
Without the windshield can headrevs just flash them?

**Comment by whateverusername0** [comment]
> Without the windshield can headrevs just flash them?

they won't be able to get revved after gaming rework

**Comment by misandrie** [comment]
> image

image

**Comment by whateverusername0** [comment]
> lower the chance for sure

ok done

**Review by Piras314 — APPROVED** [review]
going to put this through an admin discussion before merge

**Review by Piras314 — CHANGES_REQUESTED** [review]
HoDs shouldn't be able to be headrevs

**Comment by JoeHammad1844** [comment]
> HoDs shouldn't be able to be headrevs

headheadrevs

**Comment by whateverusername0** [comment]
> HoDs shouldn't be able to be headrevs

sure

**Comment by Piras314** [comment]
PR this to the Halloween-2024 branch @whateverusername0 #714

**Pull request closed without merging**